### PR TITLE
Evalg 81 - Adding content filtering C# 

### PR DESCRIPTION
**Pull request by xFrenchy**
### Summary
EVALG-81

### Details and comments
Adding content filtering C# to tutorial

### Checks

-   [x] I have updated the documentation accordingly.
-   [x] I have read the [CONTRIBUTING](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/CONTRIBUTING.md) document.

Questions:

- For the Optional exercise of updating the filter, I added a third command line argument. (pub, sub, filter). Is this a good assumption?
- We state that the filter can now be changed at runtime, but the example doesn't truly show that. The filter gets changed to "Kitchen" and is only changed once, artificially. Should we update it so that the user gets to decide _when_ a change happens and to _what_
- When creating the parameters, in C++11 you used a StringSeq. In C# it seems that I should have used a Rti.Types.Sequence<String> but that is equivalent to a native List. So I used a List, is that alright?
- In the README, I state how to run the example, however in C++ README, I see that it simply says to see the tutorial for instructions. The tutorial doesn't have C# yet so I wasn't sure if I should repeat the same instructions (although it would be missing) or to add them to the README for now and update later?

### CODE REVIEW

File: tutorials/content_filtering/cs/NuGet.Config
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/NuGet.Config:1]: Delete this file. It's generated by rtiddsgen.

File: tutorials/content_filtering/cs/HomeAutomationSubscriber_update_filter.cs
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/HomeAutomationSubscriber_update_filter.cs:1]: Use C# naming conventions: HomeAutomationSubscriberUpdateFilter.cs

File: tutorials/content_filtering/cs/HomeAutomationSubscriber_update_filter.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Rti.Dds.Core;
+using Rti.Dds.Domain;
+using Rti.Dds.Subscription;
+using Rti.Dds.Topics;
+
+namespace HomeAutomation
+{
+    public class Subscriber_Update_Filter
```
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/HomeAutomationSubscriber_update_filter.cs:24]:
```suggestion
    public class SubscriberUpdateFilter
```

File: tutorials/content_filtering/cs/HomeAutomationSubscriber_update_filter.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            using DomainParticipant participant =
+                DomainParticipantFactory.Instance.CreateParticipant(domainId: 0);
+
+            Topic<DeviceStatus> topic =
+                participant.CreateTopic<DeviceStatus>("WindowStatus");
+
+            List<String> parameters = new List<String>();
+            parameters.Add("'LivingRoom'");
+
+            ContentFilteredTopic<DeviceStatus> content_filtered_topic =
```
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/HomeAutomationSubscriber_update_filter.cs:37]:
```suggestion
            ContentFilteredTopic<DeviceStatus> contentFilteredTopic = 
```

File: tutorials/content_filtering/cs/HomeAutomationSubscriber.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        public static async Task MonitorSensors()
+        {
+            using DomainParticipant participant =
+                DomainParticipantFactory.Instance.CreateParticipant(domainId: 0);
+
+            Topic<DeviceStatus> topic =
+                participant.CreateTopic<DeviceStatus>("WindowStatus");
+
+            ContentFilteredTopic<DeviceStatus> content_filtered_topic =
```
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/HomeAutomationSubscriber.cs:33]:
```suggestion
            ContentFilteredTopic<DeviceStatus> contentFilteredTopic = 
```

(C# naming conventions)

File: tutorials/content_filtering/cs/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+Running subscriber
+
+```sh
+dotnet run -- sub
+...
+```
+
+Running filter update
```
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/README.md:42]:
```suggestion
(Optional) Run the subscriber that updates the filter
```

File: tutorials/content_filtering/cs/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+2.  Build the applications:
+
+```sh
+dotnet build
+```
+
+## Running the Example :rocket:
+
+Running publisher
```
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/README.md:28]:
```suggestion
Run the publisher:
```

Would you mind updating the READMEs for the other C# examples from "Running X" to "Run the X:" as well?

File: tutorials/content_filtering/cs/HomeAutomationSubscriber_update_filter.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+            DataReader<DeviceStatus> reader = 
+                participant.ImplicitSubscriber.CreateDataReader(content_filtered_topic);
+
+            parameters[0] = "'Kitchen'";
+            content_filtered_topic.FilterParameters = parameters;
+
+            await foreach (var data in reader.TakeAsync().ValidData())
+            {
+                if (data.is_open)
```
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/HomeAutomationSubscriber_update_filter.cs:51]: Delete the if. In the exercise, we show that the filtering is now done by the CFT, so all samples are guaranteed to have `is_open` set to true.

File: tutorials/content_filtering/cs/HomeAutomationSubscriber.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                    "FilterRoomAndOpenWindows",
+                    topic,
+                    new Filter("is_open = true and room_name = 'LivingRoom'"));
+
+            DataReader<DeviceStatus> reader = 
+                participant.ImplicitSubscriber.CreateDataReader(content_filtered_topic);
+
+            await foreach (var data in reader.TakeAsync().ValidData())
+            {
+                if (data.is_open)
```
Comment (alexcamposruiz) [inline tutorials/content_filtering/cs/HomeAutomationSubscriber.cs:44]: Delete the if. In the exercise, we show that the filtering is now done by the CFT, so all samples are guaranteed to have is_open set to true.

**Review by alexcamposruiz — CHANGES_REQUESTED** [review]

File: examples/connext_secure/certificate_revocation_list/c++11/USER_QOS_PROFILES.xml [hunk 8]
```diff
@@ -52,7 +52,7 @@ to use the software. -->
               </element>
               <element>
                 <name>com.rti.serv.secure.files_poll_interval</name>
-                <value>5000</value>
+                <value>5</value>
```
Comment (alexcamposruiz) [inline examples/connext_secure/certificate_revocation_list/c++11/USER_QOS_PROFILES.xml:55]: Is this change related to your PR at all?

Comment (xFrenchy) [reply examples/connext_secure/certificate_revocation_list/c++11/USER_QOS_PROFILES.xml:55]: No it is not. I opened a different PR in April for this change. I discovered this when Support was doing "Training as One" to learn about new features and this example was broken. https://github.com/rticommunity/rticonnextdds-examples/pull/684

File: tutorials/content_filtering/cs/README.md (hunk 5 above)
Comment (xFrenchy) [reply tutorials/content_filtering/cs/README.md]: I updated the other c# related READMEs. I noticed that the optional things are marked: Extra. Run the ...

Would you like me to change that as well to: (Optional) Run the ...

**Review by alexcamposruiz — DISMISSED** [review]

**Review by alexcamposruiz — APPROVED** [review]
